src/js/tracker.js

```diff
@@ -1604,25 +1562,60 @@
 				helpers.addEventListener(windowAlias, 'resize', activityHandler);
 				helpers.addEventListener(windowAlias, 'focus', activityHandler);
 				helpers.addEventListener(windowAlias, 'blur', activityHandler);
+			}
 
+			if (resetActivityTrackingOnPageView || ((activityTrackingConfig.activityTrackingEnabled || activityTrackingCallbackConfig.activityTrackingEnabled) && installingActivityTracking)) {
 				// Periodic check for activity.
 				lastActivityTime = now.getTime();
-				clearInterval(pagePingInterval);
-				pagePingInterval = setInterval(function heartBeat() {
-					var now = new Date();
-
-					// There was activity during the heart beat period;
-					// on average, this is going to overstate the visitDuration by configHeartBeatTimer/2
-					if ((lastActivityTime + configHeartBeatTimer) > now.getTime()) {
-						// Send ping if minimum visit time has elapsed
-						if (configMinimumVisitTime < now.getTime()) {
-							logPagePing(finalizeContexts(context, contextCallback)); // Grab the min/max globals
-						}
-					}
-				}, configHeartBeatTimer);
+
+				//Clear page ping heartbeat on new page view
+				if (pagePingInterval !== null) {
+					clearTimeout(pagePingInterval);
+					pagePingInterval = null;
+				}
+
+				if (activityTrackingConfig.activityTrackingEnabled && activityTrackingInstalled) {
+					pagePingInterval = activityInterval({
+						...activityTrackingConfig,
+						configMinimumVisitTime: lastActivityTime + activityTrackingConfig.configMinimumVisitLength * 1000,
+						callback: args => logPagePing({ context: finalizeContexts(context, contextCallback), ...args }) // Grab the min/max globals
+					});
+				}
+
+				//Clear page ping heartbeat on new page view
+				if (pagePingCallbackInterval !== null) {
+					clearTimeout(pagePingCallbackInterval);
+					pagePingCallbackInterval = null;
+				}
+
+				if (activityTrackingCallbackConfig.activityTrackingEnabled && activityTrackingInstalled) {
+					pagePingCallbackInterval = activityInterval({
+						...activityTrackingCallbackConfig,
+						configMinimumVisitTime: lastActivityTime + activityTrackingCallbackConfig.configMinimumVisitLength * 1000
+					});
+				}
 			}
 		}
 
+		function activityInterval({ activityTrackingEnabled, configHeartBeatTimer, configMinimumVisitTime, callback }) {
+			if (!activityTrackingEnabled) return;
```

Comment:
So here you `return` nothing, which basically sets pagePingInterval to `undefined`.  But elsewhere in the code you are checking `pagePingInterval === null` (not undefined).  I think you are safe for now, but this might be a null/undefined bug waiting to happen in future.

You could just drop the null equality check at line 1572, because I think it safe to do `clearTimeout(null)`.

Reply:
Great spot!
In fact, the early return can go, we're never going to end up inside this function if `activityTrackingEnabled` is `false`. It's called in two places currently and both check for this being true before calling activityInterval.